### DISCUSSION TITLE
Front-end registration spam

### DISCUSSION THREAD

**Discussion by bluestormdesign** [craftcms/cms#15858] [Q&A]
We have a front-end registration template in one of our projects and a custom event hook that sends an email out every time someone registers. 

Is there a way to battle bots spamming front-end registration forms? I don't think this is natively supported and couple of posts I've seen online mentioned custom JS captcha injection as registration is AJAX based, however it's not a safe or recommended solution.

Does anyone have any ideas how to tackle this?

**Comment by mandrasch** [comment]
Hi! You could checkout https://plugins.craftcms.com/captcha-eu?craft5= by https://www.captcha.eu/ ($)

Only used it for Formie forms, but see this setting:

<img width="326" alt="image" src="https://github.com/user-attachments/assets/fe2d1974-b293-46e5-9f67-cdd7ba0904ca">

You said: 

> however it's not a safe or recommended solution

Where did you find this statement?

There are plugins like

- https://plugins.craftcms.com/snaptcha?craft5=
- https://plugins.craftcms.com/turnstile?craft5=
- https://plugins.craftcms.com/friendly-captcha?craft5=
- https://plugins.craftcms.com/craft-hcaptcha?craft5=
- https://plugins.craftcms.com/craft-recaptcha?craft5

They inject stuff into the forms as well (by letting you adding code to twig) - haven't checked if they support the registration form. But since the registration form is **your** twig code in the end, you could inject any captcha solution you see fit?

Much success!
